skrub/_drop_uninformative.py

```diff
@@ -0,0 +1,172 @@
+import numbers
+
+from sklearn.utils.validation import check_is_fitted
+
+from . import _dataframe as sbd
+from ._on_each_column import SingleColumnTransformer
+
+__all__ = ["DropUninformative"]
+
+
+class DropUninformative(SingleColumnTransformer):
+    """Drop column if it is found to be uninformative according to various criteria.
+
+    A column is considered to be "uninformative" if one or more of the following
+    issues are found:
+
+    - The fraction of missing values is larger than a certain fraction (by default,
+      all values must be null for the column to be dropped).
+    - The column includes only one unique value (the column is constant). Missing
+      values are considered a separate value.
+    - The number of unique values in the column is equal to the length of the
+      column.
+
+    Parameters
+    ----------
+    drop_if_constant : bool, default=False
+        If True, drop the column if it contains only one unique value. Missing values
+        count as one additional distinct value.
+
+    drop_if_id : bool, default=False
```

Comment:
What about `drop_if_unique`? As we have no way to identify IDs, `drop_if_id` could be a little misleading? WDYT? Also, it could be nice to add an explainer saying that free-form text is likely to be dropped too, so to be used with caution